clang/lib/Sema/BoundsAnalysis.cpp

```diff
@@ -752,10 +751,29 @@ BoundsMapTy BoundsAnalysis::GetWidenedBounds(const CFGBlock *B) {
   return EB->In;
 }
 
+Expr *BoundsAnalysis::GetTerminatorCondition(const Expr *E) const {
+  if (!dyn_cast<BinaryOperator>(E->IgnoreParens())){
```

Comment:
You can also get rid of this check and move the `if (const auto *BO = dyn_cast<BinaryOperator>` check here.